core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigDiagnostic.java

```diff
@@ -91,28 +92,46 @@ public static void unknown(NameIterator name) {
      * @param properties the set of possible unused properties
      */
     public static void unknownProperties(Set<String> properties) {
+        if (properties.isEmpty()) {
+            return;
+        }
         SmallRyeConfig config = ConfigProvider.getConfig().unwrap(SmallRyeConfig.class);
         Set<String> usedProperties = new HashSet<>();
+        StringBuilder tmp = null;
         for (String property : config.getPropertyNames()) {
             if (properties.contains(property)) {
                 continue;
             }
-
-            usedProperties.add(StringUtil.replaceNonAlphanumericByUnderscores(property));
+            if (tmp == null) {
+                tmp = new StringBuilder(property.length());
+            } else {
+                tmp.setLength(0);
+            }
+            String usedProperty = StringUtil.replaceNonAlphanumericByUnderscores(property, tmp);
```

Comment:
What's the purpose of it exactly? I would have thought it could be to match env variables but we don't uppercase the string so i'm a bit confused.

Reply:
I didn't changed what the original code was doing TBH, so...we should understand first why it was doing before this change...and maybe @radcortez knows it more...

Reply:
We have some better ways to check this now. It was in my TODO list to rewrite it :)

Reply:
Ah yes, sorry, I meant to ping @radcortez but forgot to prepend my comment with it!